File: client2/src/game/components/platform.ts
import { ComponentFlag } from '../flags'

export class Platform {
  static _id = 'Platform'
  _id = Platform._id

  static _flag = ComponentFlag.Platform
  _flag = ComponentFlag.Platform

  fall: number
  initialX: number
  initialY: number

  constructor(fall: number = 0, initialX: number = 0, initialY: number = 0) {
    this.fall = fall
    this.initialX = initialX
    this.initialY = initialY
  }
}

File: archive/client/src/game/components/platform.lua
local constants = require 'src.game.components.constants'

local Platform = {
  _meta = constants.Platform
}

function Platform.new(id, fall, initialX, initialY)
  return {
    _meta = Platform._meta,
    id = id,
    fall = fall or 0, -- time after contact that platform will fall
    initialX = initialX or 0,
    initialY = initialY or 0
  }
end

return Platform
